src/components/v2/values/chartValuesDiff/ChartValuesView.tsx

```diff
@@ -453,24 +456,40 @@ function ChartValuesView({
     }
 
     const handleRepoChartValueChange = (event) => {
-        dispatch({ type: ChartValuesViewActionTypes.repoChartValue, payload: event })
-
         if (isExternalApp) {
-            getChartValuesList(event.chartId, (_chartValuesList: ChartValuesType[]) => {
-                if (!commonState.installedAppInfo) {
-                    const _defaultChartValues: ChartValuesType = {
-                        id: 0,
-                        kind: ChartKind.EXISTING,
-                        name: commonState.releaseInfo.deployedAppDetail.appName,
-                    }
-
-                    _chartValuesList?.push(_defaultChartValues)
-                    handleChartValuesSelection(_defaultChartValues)
-                }
-                setChartValuesList(_chartValuesList)
+            dispatch({
+                type: ChartValuesViewActionTypes.multipleOptions,
+                payload: {
+                    repoChartValue: event ?? {
+                        appStoreApplicationVersionId: 0,
+                        chartRepoName: '',
+                        chartId: 0,
+                        chartName: commonState.releaseInfo.deployedAppDetail.chartName,
+                        version: commonState.releaseInfo.deployedAppDetail.chartVersion,
+                        deprecated: false,
+                    },
+                    showConnectToChartTippy: false,
+                },
             })
-            fetchChartVersionsData(event.chartId, dispatch, commonState.releaseInfo.deployedAppDetail.chartVersion)
+
+            if (event) {
+                getChartValuesList(event.chartId, (_chartValuesList: ChartValuesType[]) => {
+                    if (!commonState.installedAppInfo) {
+                        const _defaultChartValues: ChartValuesType = {
+                            id: 0,
+                            kind: ChartKind.EXISTING,
+                            name: commonState.releaseInfo.deployedAppDetail.appName,
+                        }
+
+                        _chartValuesList?.push(_defaultChartValues)
+                        handleChartValuesSelection(_defaultChartValues)
+                    }
+                    setChartValuesList(_chartValuesList)
+                })
+                fetchChartVersionsData(event.chartId, dispatch, commonState.releaseInfo.deployedAppDetail.chartVersion)
+            }
         } else {
+            dispatch({ type: ChartValuesViewActionTypes.repoChartValue, payload: event })
```

Comment:
The payload is expected to be null or undefined?

Reply:
event check was needed in the case of an external app only.